iotics/lib/identity/api/advanced_api.py

```diff
@@ -713,18 +713,31 @@ def delegate_authentication(self, delegating_secrets: KeyPairSecrets, delegating
         subject_issuer, proof = AdvancedIdentityLocalApi.create_delegation_proof(delegating_issuer,
                                                                                  subject_doc,
                                                                                  subject_secrets)
+
+        existing_delegation = RegisterDocumentHelper.get_register_delegation_proof_by_controller(subject_issuer,
+                                                                                                 delegating_doc,
+                                                                                                 True)
+        if existing_delegation and (
+            not delegation_name or delegation_name == existing_delegation.name
```

Comment:
`delegation_name is None`? (is empty string validated against later - in which case better explicit here maybe?)

Reply:
Looks like `RegisterDelegationProof.build` will validate any `str` of this

Reply:
This was deliberate. Names must be at least "#" + one character so "" should raise a validation error at document build time

Reply:
> should raise a validation error at document build time

Ok, so the sight change here is that if the user explicitly specifies an empty string, it will now generate a name whereas before you'd get an error, right? (And that might well be fine - just checking.)

Reply:
Yes